stellargraph/mapper/sequences.py

```diff
@@ -527,3 +530,128 @@ def __len__(self):
 
     def __getitem__(self, index):
         return self.inputs, self.targets
+
+
+@experimental(reason="Missing unit tests.")
+class GraphSequence(Sequence):
+    """
+    A Keras-compatible data generator for training and evaluating graph classification models.
+    Use this class with the Keras methods :meth:`keras.Model.fit`,
+        :meth:`keras.Model.evaluate`, and
+        :meth:`keras.Model.predict`,
+
+    This class should be created using the `.flow(...)` method of
+    :class:`GraphGenerator`.
+
+    Args:
+        graphs (list)): The graphs as StellarGraph objects.
+        targets (np.ndarray, optional): An optional array of graph targets of size (N x C),
+            where N is the number of graphs and C is the target size (e.g., number of classes.)
+        normalize_adj (bool, optional): Specifies whether the adjacency matrix for each graph should
+            be normalized or not. The default is True.
+        batch_size (int, optional): The batch size. It defaults to 1.
+        name (str, optional): An optional name for this generator object.
+    """
+
+    def __init__(
+        self, graphs, targets=None, normalize_adj=True, batch_size=1, name=None
+    ):
+
+        self.name = name
+        self.graphs = graphs
+        self.normalize_adj = normalize_adj
+        self.targets = targets
+        self.batch_size = batch_size
+        # we assume that all graphs have node features of the same dimensionality
+        self.node_features_size = graphs[0].node_features(graphs[0].nodes()).shape[1]
+
+        if targets is not None:
+            if len(graphs) != len(targets):
+                raise ValueError(
+                    "When passed together targets and the number of graphs should be the same length."
+                )
+
+            self.targets = np.asanyarray(targets)
+
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.graphs) / self.batch_size))
+
+    def __normalize_adj(self, adj):
```

Comment:
This looks quite similar to the `normalize_adj` function in `core.utils`:

https://github.com/stellargraph/stellargraph/blob/cbbe37b67f11d28a265165d19644ad4789a2a8c2/stellargraph/core/utils.py#L35-L53

Could you explain the reason for reimplementing it?

Reply:
I didn't know we had this in utils. I'll check if it is the same op and replace it.